CI Failing

This fails:
```
./gradlew leakcanary-android-instrumentation:connectedCheck
```

On API 16. 

> java.lang.AssertionError: Expected exactly one leak in HeapAnalysisSuccess(heapDumpFile=/data/data/com.squareup.leakcanary.instrumentation.test/files/instrumentation_tests_heapdump.hprof, createdAtTimeMillis=1559320564745, analysisDurationMillis=40357, retainedInstances=[NoPathToInstance(referenceKey=7d46428f-e866-4471-8510-cc0e1b2e1192, referenceName=, instanceClassName=android.view.View, watchDurationMillis=5193)])


FragmentLeakTest


fragmentShouldLeak | passed (25.483s)
fragmentViewShouldLeak | failed (52.035s)

--- a/leakcanary-haha/src/main/java/leakcanary/HprofParser.kt
+++ b/leakcanary-haha/src/main/java/leakcanary/HprofParser.kt
@@ -413,7 +413,13 @@ class HprofParser private constructor(
                 if (callback != null || computeObjectClassSize) {
                   val classDumpRecord = readClassDumpRecord(id)
                   if (computeObjectClassSize) {
-                    maybeEmptyInstancesAreEmpty = when (classDumpRecord.instanceSize) {
+                    // In Android 16 classDumpRecord.instanceSize can be 8 yet there are 0 fields.
+                    // Better rely on our own computation of instance size.
+                    // See #1374
+                    val objectClassFieldSize = classDumpRecord.fields.sumBy {
+                      typeSize(it.type)
+                    }
+                    maybeEmptyInstancesAreEmpty = when (objectClassFieldSize) {
                       0 -> false
                       maybeEmptySize -> true
                       else ->

